src/cogent3/core/new_moltype.py

```diff
@@ -647,18 +647,21 @@ def _(self, seq: str, validate: bool = True) -> bool:
                 f"{seq[:4]!r} not valid for moltype {self.name!r}"
             )
         return self.is_degenerate(
-            self.degen_gapped_alphabet.to_indices(seq), validate=False
+            self.most_degen_alphabet().to_indices(seq), validate=False
         )
 
     @is_degenerate.register
     def _(self, seq: numpy.ndarray, validate: bool = True) -> bool:
-        # what index is the first degenerate character
+        if self.degen_alphabet is None:
+            return False
+
         if validate and not self.is_valid(seq):
             raise new_alphabet.AlphabetError(
                 f"{seq[:4]!r} not valid for moltype {self.name!r}"
             )
 
-        for index, val in enumerate(self.degen_gapped_alphabet):
+        # what index is the first degenerate character
```

Comment:
```python
# since you've already excluded non-degen moltypes
first_degen = self.most_degen_alphabet().gap_index + 1
```

Reply:
@GavinHuttley That's not a question. That is a comment (that already existed in the function) which I moved below my changes. 

The code below the comment finds the first degenerate character - which I can update to the snippet you provided above if you'd prefer

Reply:
my approach more succinct, try that and let me know if it fails!

Reply:
it works!